### DISCUSSION TITLE
"Category" field of "Create Part" might has wrong mapping.

### DISCUSSION THREAD

**Discussion by Century1026** [inventree/InvenTree#7318] [Q&A] [answered]
I first discovered the problem when the title "Category" was not able to be translated.

Then I made changes on title names and help text.
![b2fb0ca92d783834732c7606abf6267](https://github.com/inventree/InvenTree/assets/106154819/27db7f5f-b47e-4fc4-9c7f-e33cd2eed69a)
![e88d4cfdac1030c87d97bfab821fb57](https://github.com/inventree/InvenTree/assets/106154819/e6e1f5ee-be87-485f-97ae-27b322a4bb5a)
However, it turned out only the "Category" field is having no changes at all.
![2d4c2348a8c4ca8caafb891273b64df](https://github.com/inventree/InvenTree/assets/106154819/31226236-b496-403c-956f-1ca873475255)
 Is the title of "Category" field defined somewhere else?

**Comment by SchrodingersGat** [comment]
@Century1026 looking into this it appears that in *some cases* the labels from the models do not get correctly passed through to the API! It will take a bit of time to dig into this, but it is definitely a bug and I will look into it.

**Reply by Century1026** [reply]
The "Minimum stock" seems not getting correctly passed through to the API as well.
![image](https://github.com/inventree/InvenTree/assets/106154819/136b67af-d536-4cb0-a774-8687fc9c7c40)
note that the second word is not capitalized, so I searched with case sensitivity and get nothing related.
It's so strange!
![image](https://github.com/inventree/InvenTree/assets/106154819/772e48af-dbf3-4d6d-afb4-fa4f8f49c351)

**Reply by SchrodingersGat** [reply, marked as answer]
@Century1026 see the PR here - https://github.com/inventree/InvenTree/pull/7343 - for more information

**Reply by Century1026** [reply]
@SchrodingersGat
I copied and pasted the new metadata.py and clicked "add part" and this happened.
![image](https://github.com/inventree/InvenTree/assets/106154819/4b48a893-f8a9-4bf9-ad6d-dfe1b33cdde1)

**Reply by Century1026** [reply]
I tried codespace with latest code
![image](https://github.com/inventree/InvenTree/assets/106154819/cfa73648-8cd0-4fdf-ba1c-86ff323f2530)
but I'm unable to login even I created superuser
![image](https://github.com/inventree/InvenTree/assets/106154819/b21a5736-86fc-4fc7-a9ba-d5b6cf3ab4a1)

**Reply by SchrodingersGat** [reply]
> @SchrodingersGat I copied and pasted the new metadata.py and clicked "add part" and this happened. 

This bug has been fixed in latest master already

**Reply by SchrodingersGat** [reply]
> I tried codespace with latest code 

There might be something else going on here, but unrelated to this issue

**Reply by Century1026** [reply]
> > @SchrodingersGat I copied and pasted the new metadata.py and clicked "add part" and this happened.
> 
> This bug has been fixed in latest master already

It worked!!!
Thank you so much, you made my day!

**Comment by Century1026** [comment]
There're still some issues related to the fields' attributes.
Provided that the attribute "blank" of both "Name" and "Category" filed is "True".
![image](https://github.com/inventree/InvenTree/assets/106154819/275f5ef9-53a5-4257-b941-2ed445ba9693)
![image](https://github.com/inventree/InvenTree/assets/106154819/b6f6e3c1-3628-438e-87dd-1fd6a969e9c9)

However, "Category" label still has "asterisk“.
![image](https://github.com/inventree/InvenTree/assets/106154819/8778b1a2-970c-4ffb-a62d-efcf60932ce9)

**Reply by SchrodingersGat** [reply]
@Century1026 thanks for this information - i have just now patched this in https://github.com/inventree/InvenTree/pull/7343

**Reply by Century1026** [reply]
@SchrodingersGat 
You're welcome! I'm glad we could work together.

**Reply by Century1026** [reply]
Hey, Oliver!
I found the "External" option is having the same issue.
The "blank=True" is not affecting anything.
![image](https://github.com/inventree/InvenTree/assets/106154819/faa55c77-43fc-45d4-a647-225b547bf0ad)
